src/SharedStream.mo

```diff
@@ -72,47 +72,77 @@ module {
 
   };
 
-  public type StreamSub<T> = ([T], Nat) -> async ();
-
-
-  /// usage:
-  /// let stream = SharedStream<Int>(10, 10, func (item) = 1);
-  /// stream.next([1, 2, 3, 4]);
-  /// stream.next([5, 6, 7, 8]);
-  /// stream.next([9, 10, 11, 12]);
-  /// stream.subscribe(func (elements: [Int]): async () {
-  ///   await anotherCanister.appendStream(elements); 
-  /// });
-  /// await* stream.emit(); // will send items 1..9 to `anotherCanister`
-  /// await* stream.emit(); // will send items 10..12 to `anotherCanister`
-  /// await* stream.emit(); // will do nothing, stream clear
-  class SharedStream<T>(
-    maxSize : Nat,
-    weightLimit : Nat,
-    weightFunc : (item : T) -> Nat,
+  /// Usage:
+  ///
+  /// let receiver = StreamReceiver<Int>(
+  ///   func (element: Int, index: Nat): () {
+  ///     ... do your logic with item
+  ///   }
+  /// );
+  ///
+  /// Hook-up receive function in the actor class:
+  /// public shared func onStreamChunk(chunk: [Int], firstIndex: Nat) : async () = async receiver.onChunk(chunk, firstIndex);
+  class StreamReceiver<T>(
+    callback : (T, Nat) -> ()
```

Comment:
We need `(T, Nat, Nat)` as arguments. We also have to pass the stream id to `callback`.

Reply:
We can make stream id a constructor argument. 

When a new stream is opened (i.e. stream id is used) then a new class instance is used.

Reply:
I think it can be managed by top-level application, because there will be one receiver per stream, like:
```let receiver1 = StreamReceiver(func (el, index) = handleStreamItem(1, el, index));
let receiver2 = StreamReceiver(func (el, index) = handleStreamItem(2, el, index));```
WDYT?

Reply:
Done